client-v2/src/shared/components/snapLink/SnapLink.tsx

```diff
@@ -133,7 +140,17 @@ const SnapLink = ({
             </SnapLinkURL>
           </CollectionItemHeadingContainer>
         </CollectionItemContent>
-        {size === 'default' && <ThumbnailSmall />}
+        <ImageWrapper>
+          <ThumbnailSmall
+            imageHide={article && article.imageHide}
+            url={article && article.imageReplace ? article.imageSrcThumb : ''}
```

Comment:
`article.imageSrcThumb` seems to give the right src for the thumbnail, but without the check on `artice.imageReplace` the thumbnail image will persist even when the toggle is switched off. Not sure if there is a better way to handle this?

Reply:
I think we're after `article.thumbnail` here, which is a property of `DerivedArticle` that displays whatever is current (taking into account overrides etc.)

Reply:
`article.thumbail` doesn't seem to contain any value, even after adding a new image from the Grid. Do we need to assign it somewhere?

Reply:
Discussed IRL -- it turns out that at the point at which a `DerivedArticle` is created, `getThumbnail` doesn't get called if an external article isn't present. We should amend accordingly 👍

Reply:
I've had a stab at that, let me know what you think.